### DISCUSSION TITLE
GPUI Documentation

### DISCUSSION THREAD

**Discussion by d3rpp** [zed-industries/zed#12334] [Feature Requests]
> [!NOTE]
>
> Before I start, I am aware that given the in-dev nature of Zed as a whole, there is not much stability in terms of APIs or implementations for GPUI. If that is the major blocker of documentation/more examples, please ignore/close this discussion.

Given how well its been utilised in Zed, I'd like to be able to use it in a Personal Project of mine, though cannot wrap my head around even the most basic of things regarding how to use it properly?

I've scoured through this code base, and am finding great difficulty in wrapping my head around it (that is a skill issue on my part). You have all shown that it is a well-made and very efficient library making what is by all means a lovely application, proving beyond all doubt this library able to accomplish a lot of things. Despite this I'm failing to connect the dots as to how it achieves this (a skill issue on my part I'm sure).

Are there any resources (current or planned) on how to effectively use GPUI, the examples directory of GPUI, while to shows how to spawn a window, leaves a lot of gaps, such as how to use it for managing component state, how to write a component, using the async runtime, etc.
 

Given this *seems* to follow something like an MVVM architecture, would people benefit from looking into the MVVM architecture of applications?

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
